be/tx.c

```diff
@@ -740,6 +740,37 @@ M0_INTERNAL void m0_be_tx_gc_enable(struct m0_be_tx *tx,
 	tx->t_gc_param   = param;
 }
 
+M0_INTERNAL void m0_be_tx_lsn_set(struct m0_be_tx *tx,
+                                  m0_bindex_t      lsn,
+                                  m0_bindex_t      lsn_discarded)
+{
+	/*
+	 * This function is called from m0_be_tx_group_close().
+	 * When the BE tx group this transaction belongs to is being closed
+	 * the tx sm group is not locked - this is why BE_TX_LOCKED_AT_STATE
+	 * could not be used here.
+	 */
+	M0_PRE(m0_be_tx_state(tx) == M0_BTS_CLOSED);
```

Comment:
M0_ENTRY() is needed.
Otherwise, no match entry for the M0_LEAVE();

Reply:
M0_ENTRY() is useless here - it doesn't give any additional information if someone is looking for a bug.

Removed it from `fdmi_int` and changed `M0_LEAVE()` to `M0_LOG(M0_DEBUG, ...)`.

Reply:
On the second thought, left M0_LEAVE() as is. It clearly states that this those are the values when the function returns.